internal/stack/logs.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stack
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/docker"
+)
+
+func dockerComposeLogs(serviceName string, snapshotFile string) ([]byte, error) {
+	p, err := compose.NewProject(DockerComposeProjectName, snapshotFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create docker compose project")
+	}
+
+	opts := compose.CommandOptions{
+		Services: []string{serviceName},
+	}
+
+	out, err := p.Logs(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "running command failed")
+	}
+	return out, nil
+}
+
+func dockerInternalLogs(serviceName string) ([]byte, bool, error) {
+	if serviceName != fleetServerService {
+		return nil, false, nil // we need to pull internal logs only from the Fleet Server container
+	}
+
+	p, err := compose.NewProject(DockerComposeProjectName)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "could not create docker compose project")
+	}
+
+	serviceContainer := p.ContainerName(serviceName)
+	out, err := docker.Exec(serviceContainer, "sh", "-c", `find data/logs/default/fleet-server-json* -printf '%p\n' -exec cat {} \;`)
```

Comment:
Would it be an option to mount the log directories as a volume, and get the logs directly from there?

Reply:
I would have to research stability of this. In general we had issues with volume mounts if directories are recreated in runtime or mounts got lost. But I like this idea, do you think we should do it here or in a followup?

There are also other files in this directory (filebeat and metricbeat logs), but we will have to store them in internal location (like GCP bucket) as they may contain confidential data.

Reply:
Sure, we can do it as a follow up.